course3/homework-list/list.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include <iterator>
+#include <memory>
+#include <initializer_list>
+#include <boost/range/adaptor/reversed.hpp>
+
+
+namespace Karun
+{
+
+template<typename T>
+class List
+{
+public:
+	typedef T value_type;
```

Comment:
No. This is the way. Actually, I personally prefer the "using" version:
```
using value_type = T;
```
Some code needs these to infer stored type (and other types) from the container type.